packages/api/src/command/medical/patient/__tests__/store-query-init.test.ts

```diff
@@ -0,0 +1,99 @@
+import { PatientModel } from "../../../../models/medical/patient";
+import { mockStartTransaction } from "../../../../models/__tests__/transaction";
+import { makePatientModel } from "../../../../models/medical/__tests__/patient";
+
+import { storeQueryInit, StoreQueryParams } from "../query-init";
+import {
+  pdParams,
+  dqParams,
+  cqParams,
+  mockedPatientAllProgresses,
+  documentQueryProgress,
+  patientDiscovery,
+  consolidatedQuery,
+} from "./store-query-cmd";
+
+let patientModel: PatientModel;
+
+let patientModel_update: jest.SpyInstance;
+let patientModel_findOne: jest.SpyInstance;
+jest.mock("../../../../models/medical/patient");
+
+beforeEach(() => {
+  patientModel = makePatientModel();
+  mockStartTransaction();
+  patientModel_findOne = jest.spyOn(PatientModel, "findOne").mockResolvedValue(patientModel);
+  patientModel_update = jest.spyOn(patientModel, "update");
+});
+
+afterEach(() => {
+  jest.restoreAllMocks();
+});
```

Comment:
Just FYI, before/after each is more expensive then before/after all. If we need to used, we need to use it. Just pointing that this should be the last option, as it results in slower tests.

Reply:
We can have the `spyOn` set on `beforeAll` and modify the return value on each test and/or on `beforeEach`. NABD though.

Reply:
Usually we can `spyOn` the class and then modify the returned/resolved value on each test and/or on `beforeEach`.

NABD though